catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -164,6 +168,37 @@ protected String generateCreateTableSql(
     return result;
   }
 
+  private Map<String, String> appendNecessaryProperties(Map<String, String> properties) {
+    Map<String, String> resultMap;
+    if (properties == null) {
+      resultMap = new HashMap<>();
+    } else {
+      resultMap = new HashMap<>(properties);
+    }
+
+    // If the backend server is less than 3, we need to set the property 'replication_num'
+    // to 1 explicitly.
+    if (!properties.containsKey(REPLICATION_FACTOR)) {
+      // Try to check the number of backend servers.
+      String query = "select count(*) from information_schema.backends where Alive = 'true'";
+
+      try (Connection connection = dataSource.getConnection();
+          Statement statement = connection.createStatement();
+          ResultSet resultSet = statement.executeQuery(query)) {
+        while (resultSet.next()) {
+          int backendCount = resultSet.getInt(1);
+          if (backendCount < 3) {
```

Comment:
More precisely, if the number of BE nodes is less than 3, you are using the default value of Doris server, not 3